cmd/gb-vendor/vendor/repo.go

```diff
@@ -212,13 +208,21 @@ type hgrepo struct {
 
 func (h *hgrepo) URL() string { return h.url }
 
-func (h *hgrepo) Clone() (WorkingCopy, error) {
+func (h *hgrepo) Checkout(branch, revision string) (WorkingCopy, error) {
 	dir, err := mktmp()
 	if err != nil {
 		return nil, err
 	}
+	args := []string{
+		"clone",
+		h.url,
+		dir,
+	}
 
-	if err := runOut(os.Stderr, "hg", "clone", h.url, dir); err != nil {
+	if branch != "" {
+		args = append(args, "--branch", branch)
+	}
+	if err := runOut(os.Stderr, "hg", args...); err != nil {
 		return nil, err
```

Comment:
Suggestion: Maybe `os.RemoveAll(dir)` when the cloning fails to cleanup the temp dir.